packages/beta/src/__tests__/api/filesMultiPart.int.spec.ts

```diff
@@ -0,0 +1,296 @@
+// Copyright 2020 Cognite AS
+import { sleepPromise } from '@cognite/sdk-core';
+import CogniteClient from '../../cogniteClient';
+import {
+  ExternalFileInfo,
+  LabelDefinition,
+  MultiPartFileChunkResponse,
+} from '../../types';
+import {
+  divideFileIntoChunks,
+  divideFileIntoStreams,
+  getFileStats,
+  toArrayBuffer,
+  setupMockableClientForIntegrationTests,
+} from '../testUtils';
+import path, { join } from 'path';
+import { FilesMultipartUploadSessionAPI } from '../../api/files/filesMultipartUploadSessionApi';
+// file to upload for integration tests
+const testfile = join(__dirname, '../VAL.nwd');
+
+describe.skip('Files: Multi part Upload Integration Tests', () => {
+  let client: CogniteClient;
+  let label: LabelDefinition;
+
+  beforeAll(async () => {
+    client = setupMockableClientForIntegrationTests();
+  });
+
+  afterAll(async () => {
+    const { items } = await client.files.list({
+      filter: {
+        labels: {
+          containsAll: [{ externalId: label.externalId }],
+        },
+      },
+    });
+    await client.files.delete(items.map((item) => ({ id: item.id })));
+    await client.labels.delete([{ externalId: label.externalId }]);
+    console.log(`cleaned ${items.length} files successfully.`);
+  });
+
+  const getFileCreateArgs = (
+    additionalFields: Partial<ExternalFileInfo> = {}
+  ) => {
+    const sourceCreatedTime = new Date();
+    const fileName = path.parse(additionalFields.name ?? 'filename_0').base;
+    const localFileMeta: ExternalFileInfo = {
+      name: fileName,
+      mimeType: 'application/octet-stream',
+      directory: '/test/testing',
+      metadata: {
+        key: 'value',
+      },
+      sourceCreatedTime,
+    };
+
+    return { sourceCreatedTime, localFileMeta };
+  };
+
+  test('can create uploadurls', async () => {
+    const response = await client.filesMultiPart.multipartUploadSession(
+      {
+        ...getFileCreateArgs({ name: testfile }).localFileMeta,
+      },
+      5,
+      false
+    );
+    expect(response.getNotCompletedParts().length).toEqual(5);
+  });
+  it.each<number>([3, 5])(
+    'can upload and get the state',
+    async (numberOfParts) => {
+      const fileChunks = divideFileIntoChunks(testfile, numberOfParts);
+      const response = await client.filesMultiPart.multipartUploadSession(
+        {
+          ...getFileCreateArgs({ name: testfile }).localFileMeta,
+        },
+        numberOfParts,
+        false
+      );
+      for (let i = 0; i < numberOfParts; i++) {
+        try {
+          await response.uploadPart(i, fileChunks[i]);
+          console.log(`Uploaded part ${i + 1} successfully.`);
+        } catch (error) {
+          console.error(`Error uploading part ${i + 1}:`, error);
+        }
+      }
+      await sleepPromise(2000);
+      const [retrievedFile] = await client.files.retrieve([
+        { id: response.multiPartFileUploadResponse.id },
+      ]);
+      expect(retrievedFile.uploaded).toBeTruthy();
```

Comment:
The check for upload=true may have to be repeated, as this is happening as part separate async processing within files API service and the specific cloud environment storage buckets.

After upload is done it normally takes less than a second before upload is set to true. 
But in certain cases can take longer time.

To avoid making this a flaky test, the check for upload=true should be retried a number of time.
If it is still not true after for example 2 minutes, then it should be ok to fail it.

Reply:
asserting with retries now